peterpy/middlewares.py

```diff
@@ -0,0 +1,31 @@
+from aiohttp import web
+import logging
+
+from peterpy.database.connection import DatabaseConnection, DatabaseSession
+from peterpy.repositories.database_product_repository import DatabaseProductRepository
+from peterpy.services.product_service import ProductService
+
+
+def db_session_wrapper_factory(db_connection: DatabaseConnection):
+    @web.middleware
+    async def db_session_wrapper(request, handler):
+        logging.debug("db_session_wrapper called")
+        engine = db_connection.engine()
+
+        with DatabaseSession(engine) as session:
+            try:
+                repository = DatabaseProductRepository(session)
```

Comment:
we want to pass the product_service to the handler. 
Because the product_service needs the repository for instantiation, and the repository needs a session, and the session is start in the middleware for every request, we create repo and service here. 
product_service is added to `request` object. 
Can't use dependency injection higher up in the app because we need the session for the repository